Firestore/core/test/firebase/firestore/testutil/testutil.h

```diff
@@ -381,8 +381,8 @@ inline core::FieldFilter Filter(absl::string_view key,
 
 inline core::FieldFilter Filter(absl::string_view key,
                                 absl::string_view op,
-                                const std::string& value) {
-  return Filter(key, op, model::FieldValue::FromString(value));
+                                const char* value) {
+  return Filter(key, op, model::FieldValue::FromString(std::string(value)));
 }
 
 inline core::FieldFilter Filter(absl::string_view key,
```

Comment:
This is actually a problematic definition. We shouldn't allow conversion from pointers to bool. The way to ensure it is to disable the overload when the type of the argument isn't an explicit bool.

The way to do that would be:

```c++
template <typename T,
          typename = typename std::enable_if<std::is_same<bool, T>{}>::type>
inline core::FieldFilter Filter(absl::string_view key,
                                absl::string_view op,
                                T bool_value) {
  return Filter(key, op, model::FieldValue::FromBoolean(bool_value));
}
```

Reply:
This is definitely something I would never have thought of, but now I get credit for it via `git blame`. Wheee.